File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09 - While Loop Exercise/06. Cake .py
width = int(input())
length = int(input())

size_cake = width * length
sum_parts = 0

while True:
    command = input()
    if command == "STOP":
        break
    part = int(command)
    sum_parts += part
    if size_cake < sum_parts:
        break

if size_cake < sum_parts:
    print(f"No more cake left! You need {sum_parts - size_cake} pieces more.")
else:
    print(f"{size_cake - sum_parts} pieces are left.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/06. Cake.js
function cake(data) {
    let size = (Number(data.shift())) * (Number(data.shift()));
    let sum = 0;

    while("I say so") {
        let part = data.shift();
        if (part === 'STOP') {break;}
        sum += Number(part);
        if (size < sum) {break;}
    }

    if (size < sum) {
        console.log(`No more cake left! You need ${sum - size} pieces more.`);
    } else {
        console.log(`${size - sum} pieces are left.`);
    }
}


cake([10, 10, 20, 20, 20, 20, 21]);  // Expected Output:
// No more cake left! You need 1 pieces more.

cake([10, 2, 2, 4, 6, 'STOP']);  // Expected Output:
// 8 pieces are left.
